zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AsterixQueryGenerator.scala

```diff
@@ -0,0 +1,270 @@
+package edu.uci.ics.cloudberry.zion.model.impl
+
+import edu.uci.ics.cloudberry.zion.model.datastore.{IQLGenerator, QueryParsingException}
+import edu.uci.ics.cloudberry.zion.model.schema._
+
+
+/**
+  * Defines constant strings for query languages supported by AsterixDB
+  */
+trait TypeImpl {
+
+  val aggregateFuncMap: Map[AggregateFunc, String]
+
+  def getAggregateStr(aggregate: AggregateFunc): String = {
+    aggregateFuncMap.get(aggregate) match {
+      case Some(impl) => impl
+      case None => throw new QueryParsingException(s"No implementation is provided for aggregate function ${aggregate.name}")
+    }
+  }
+
+  val datetime: String = "datetime"
+  val round: String = "round"
```

Comment:
Here AQL and SQL++ actually use the same implementation of datetime and round, thus their implementations are extracted to AsterixImpl.
If we move them to the companion object, then we have to use AsterixImpl.datetime and AsterixImpl.round to use them, which are not compatible with other fields...